src/components/Charts/PriceChart/utils.ts

```diff
@@ -0,0 +1,42 @@
+import { timeDay, timeHour, TimeInterval, timeMinute, timeMonth } from 'd3'
+import { TimePeriod } from 'graphql/data/util'
+
+const fiveMinutes = timeMinute.every(5)
+// Since 5 is valid input for timeMinute.every, fiveMinutes should never be null
+if (fiveMinutes === null) {
+  throw new Error('Invalid minute interval')
+}
+
+const TIME_PERIOD_INTERVAL_TABLE: Record<TimePeriod, { interval: TimeInterval; step: number }> = {
+  [TimePeriod.HOUR]: { interval: fiveMinutes, step: 2 }, // spaced 10 minutes apart at times that end in 0 or 5
+  [TimePeriod.DAY]: { interval: timeHour, step: 4 }, // spaced 4 hours apart
+  [TimePeriod.WEEK]: { interval: timeDay, step: 1 }, // spaced 1 day apart
+  [TimePeriod.MONTH]: { interval: timeDay, step: 7 }, // spaced 1 week apart
+  [TimePeriod.YEAR]: { interval: timeMonth, step: 2 }, // spaced 2 months apart
+}
+
+/**
+ * Returns an array of tick values for a given time range and time period.
+ * This function makes sure that the ticks are evenly spaced and are not too close to the edges.
+ */
+export function getTicks(startTime: number, endTime: number, timePeriod: TimePeriod, maxTicks: number) {
+  if (maxTicks === 0 || endTime <= startTime) return []
+
+  // Prevents ticks from being too close to the axis edge
+  const tickMargin = (endTime - startTime) / 24
+
+  const startDate = new Date((startTime + tickMargin) * 1000)
+  const endDate = new Date((endTime - tickMargin) * 1000)
+
+  const { interval, step } = TIME_PERIOD_INTERVAL_TABLE[timePeriod]
+  const ticks = interval.range(startDate, endDate, step).map((x) => x.valueOf() / 1000) // convert to seconds
+
+  if (ticks.length <= maxTicks) return ticks
+
+  const newTicks = []
+  const tickSpacing = Math.floor(ticks.length / maxTicks)
+  for (let i = 1; i < ticks.length; i += tickSpacing) {
+    newTicks.push(ticks[i])
+  }
```

Comment:
i don't think this is doing exactly what you want it to do. so you know the `tickSpacing` is actually `1/maxTicks` right. so what you actually want to do is increment by `1/maxTicks` and only choose one tick that is inside that area so something like this:

```
const tickSpacing = 1 / maxTicks
for (let i = 0; i < 1; i += tickSpacing) {
  const maxTick = Math.floor(i * ticks.length)
  newTicks.push(ticks[maxTick])
}
```

Reply:
this code was copied over from the existing solution, I think it's working as originally intended

tickSpacing currently represents how many (original) ticks apart the new ticks should be. what would 1/maxTicks represent? im confused why we would increment by that fraction in the for loop

Reply:
so with the old one if `ticks.length` is 1.9x the size of `maxTicks` the ticksSpacing is `1` so you are just basically getting the first `maxTicks` of `ticks`. If you do it by the way of using `1/maxTicks` thats basically the percentage of the `maxTicks` that you want to map `ticks` onto so you are able to graph the data across the whole subset of `ticks` instead of the first `maxTicks` of `ticks`

Reply:
current way: `maxTicks = 1000` `ticks.length = 1900` `newTicks = ticks[0...999]`
proposed way: `maxTicks = 1000` `ticks.length = 1900` `newTicks = ticks[0...1900]` where we grab every other 1.9 tick

Reply:
Ahhh okay I think I understand what you're saying. I think the discrete set of ticks.length and maxTicks that this function currently deals with don't produce an edge case like this. I think since this is a refactor PR the functionality shouldn't change for now, but we can improve in the future esp if we have larger charts or more ticks.

One issue tho, I think your solution is more inclusive of all items in the original array, but doesn't produce evenly spaced ticks

<img width="492" alt="image" src="https://github.com/Uniswap/interface/assets/39385577/36f8d629-a959-4132-8292-7be385458126">

Reply:
Ah true all good then